lightblue/src/test/java/org/esbtools/eventhandler/lightblue/testing/MultiStringDocumentEvent.java

```diff
@@ -116,7 +119,8 @@ public boolean isSupersededBy(DocumentEvent event) {
 
     @Override
     public boolean couldMergeWith(DocumentEvent event) {
-        return event instanceof MultiStringDocumentEvent;
+        return event instanceof MultiStringDocumentEvent &&
+                !event.isSupersededBy(this);
```

Comment:
Methinks this second check is technically unnecessary... since `LightblueDocumentEventRepository.addEvent` does the same check before ever calling `couldMergeWith`. I'm not suggesting changing it. I just found this interesting.